src/pages/api/identities/[id].ts

```diff
@@ -31,17 +34,80 @@ import { sentryMiddleware } from "@src/middlewares/sentry-middleware";
 import { generateAlertMessage } from "@src/utils/generate-alert-message";
 import { getIdentityType } from "@src/utils/get-identity-type";
 
-const handler: Handler = (request, response): Promise<void> => {
+const get: Handler = async (request, response) => {
+  const webErrors = {
+    identityNotFound: ERRORS_DATA.IDENTITY_NOT_FOUND,
+    graphqlErrors: ERRORS_DATA.GRAPHQL_ERRORS,
+    connectUnreachable: ERRORS_DATA.CONNECT_UNREACHABLE,
+    invalidQueryString: ERRORS_DATA.INVALID_QUERY_STRING,
+  };
+
+  return getTracer().span("get-identity handler", async (span) => {
+    const userCookie = await getServerSideCookies<UserCookie>(request, {
+      cookieName: "user-cookie",
+      isCookieSealed: true,
+      cookieSalt: configVariables.cookieSalt,
+    });
+
+    if (!userCookie) {
+      response.statusCode = HttpStatus.TEMPORARY_REDIRECT;
+      response.setHeader("location", "/");
+      response.end();
+      return;
+    }
+
+    const { id: identityId } = request.query;
+
+    if (!identityId || typeof identityId !== "string") {
+      throw webErrorFactory(webErrors.invalidQueryString);
+    }
+
+    const identity = await getIdentity(configVariables.managementCredentials, {
+      userId: userCookie.sub,
+      identityId,
+    }).catch((error) => {
+      span.setDisclosedAttribute("is Identity fetched", false);
+
+      if (error instanceof IdentityNotFoundError) {
+        throw webErrorFactory({
+          ...webErrors.identityNotFound,
+        });
+      }
+
+      if (error instanceof GraphqlErrors) {
+        throw webErrorFactory({
+          ...webErrors.graphqlErrors,
+          parentError: error,
+        });
+      }
+
+      if (error instanceof ConnectUnreachableError) {
+        throw webErrorFactory({
+          ...webErrors.connectUnreachable,
+          parentError: error,
+        });
+      }
+
+      throw error;
+    });
+
+    span.setDisclosedAttribute("is Identity fetched", true);
+    response.statusCode = HttpStatus.OK;
+    response.json(identity);
+  });
+};
+
+const destroy: Handler = (request, response): Promise<void> => {
   const webErrors = {
     badRequest: ERRORS_DATA.BAD_REQUEST,
     identityNotFound: ERRORS_DATA.IDENTITY_NOT_FOUND,
     connectUnreachable: ERRORS_DATA.CONNECT_UNREACHABLE,
   };
 
   return getTracer().span("delete-identity handler", async (span) => {
-    const { type, value } = request.body;
+    const { type, value, id } = request.body;
 
-    if (!type || !value) {
+    if (!type || !value || id !== request.query.id) {
```

Comment:
I just verify if the id in the url params is the same as the one in the body. Its just "one more" check as Fenn said to me